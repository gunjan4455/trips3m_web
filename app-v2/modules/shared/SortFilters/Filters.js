import React from 'react';
import PropTypes from 'prop-types';

import { FilterIcon } from 'helpers/Icon/Icon';
import './SortAndFilter.scss';

const Filters = ({ selectedFilters }) => {
  const calculateFiltersCount = (filters) => {
    const selectedFilters = filters;
    let _selectedFilterCount = 0;
    Object.keys(selectedFilters).map((filter) => {
      Object.keys(selectedFilters[filter]).map((filterValue) => {
        if (selectedFilters[filter][filterValue]) {
          _selectedFilterCount += 1;
        }
        return 0;
      });
      return 0;
    });
    return _selectedFilterCount;
  };

  return (
    <div className="row row-">
      <div className="relative pl48 filter-by-listing-view">
        <span className="sort-by-icon"><FilterIcon /></span>
        <p>Filters</p>
        <p className={`f10 no-filter-selected`}>{calculateFiltersCount(selectedFilters)} selected</p>
      </div>
    </div>
  );
};

Filters.propTypes = {
  selectedFilters: PropTypes.object
};

Filters.defaultProps = {
  selectedFilters: {}
};

export default Filters;
