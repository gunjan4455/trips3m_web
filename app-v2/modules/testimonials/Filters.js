import React from 'react';
import PropTypes from 'prop-types';

import { FilterIcon } from 'helpers/Icon/Icon';

const Filters = ({ initialFilters }) => {
  return (
    <div>
      <div className="pl15 pb5 pt5 pr15 flex alignCenter">
        <span className="iblock input-24 verticalMiddle">
          <FilterIcon/>
        </span>
        <div className="iblock pl15 verticalMiddle">
          <p className="iblock f14 fw4 pfc3 pb2">Filters</p>
          <p className="at_filterlength fw4 f10 pfc4">{initialFilters.length} selected</p>
        </div>
      </div>
    </div>
  );
};

Filters.propTypes = {
  initialFilters: PropTypes.array,
};

Filters.defaultProps = {
  initialFilters: [],
};

export default Filters;

