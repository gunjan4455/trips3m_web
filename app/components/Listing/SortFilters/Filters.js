import React from 'react';
import PropTypes from 'prop-types';
import {
  FilterIcon
} from 'helpers/Icon/Icon';


import './SortAndFilter.scss';

const SortBy = ({ numberOfFilterSelected }) => {
  const filterClass = (numberOfFilterSelected() ? 'selected-filters' : '');
  return (
    <div className="row row-">
      <div className="relative pl48 filter-by-listing-view" >
        <span className="sort-by-icon"><FilterIcon /></span>
        <p>Filters</p>
        <p className={`f10 no-filter-selected ${filterClass}`}>{numberOfFilterSelected()} selected</p>
      </div>
    </div>
  );
};

SortBy.propTypes = {
  numberOfFilterSelected: PropTypes.func
};

SortBy.defaultProps = {
  numberOfFilterSelected: () => {}
};
export default SortBy;
