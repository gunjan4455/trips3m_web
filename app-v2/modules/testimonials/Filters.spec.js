import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Filters from './Filters';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  initialFilters: [],
};

const shallowWrapperFilter = props => {
  return shallow(<Filters {...props} />);
};

describe('Testimonials module', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperFilter(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render proper filter length', () => {
    const wrapper = shallowWrapperFilter(props);
    expect(wrapper.find('.at_filterlength').text()).toEqual('0 selected');
  });

  it('should render proper filter length', () => {
    props.initialFilters.push(4);
    const wrapper = shallowWrapperFilter(props);
    expect(wrapper.find('.at_filterlength').text()).toEqual('1 selected');
  });
});
