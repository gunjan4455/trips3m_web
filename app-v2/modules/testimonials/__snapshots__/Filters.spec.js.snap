// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filters
    initialFilters={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="pl15 pb5 pt5 pr15 flex alignCenter"
      >
        <span
          className="iblock input-24 verticalMiddle"
        >
          <IconSvg />
        </span>
        <div
          className="iblock pl15 verticalMiddle"
        >
          <p
            className="iblock f14 fw4 pfc3 pb2"
          >
            Filters
          </p>
          <p
            className="at_filterlength fw4 f10 pfc4"
          >
            0
             selected
          </p>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="iblock input-24 verticalMiddle"
          >
            <IconSvg />
          </span>,
          <div
            className="iblock pl15 verticalMiddle"
          >
            <p
              className="iblock f14 fw4 pfc3 pb2"
            >
              Filters
            </p>
            <p
              className="at_filterlength fw4 f10 pfc4"
            >
              0
               selected
            </p>
          </div>,
        ],
        "className": "pl15 pb5 pt5 pr15 flex alignCenter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "iblock input-24 verticalMiddle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="iblock f14 fw4 pfc3 pb2"
              >
                Filters
              </p>,
              <p
                className="at_filterlength fw4 f10 pfc4"
              >
                0
                 selected
              </p>,
            ],
            "className": "iblock pl15 verticalMiddle",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Filters",
                "className": "iblock f14 fw4 pfc3 pb2",
              },
              "ref": null,
              "rendered": "Filters",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  0,
                  " selected",
                ],
                "className": "at_filterlength fw4 f10 pfc4",
              },
              "ref": null,
              "rendered": Array [
                0,
                " selected",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="pl15 pb5 pt5 pr15 flex alignCenter"
        >
          <span
            className="iblock input-24 verticalMiddle"
          >
            <IconSvg />
          </span>
          <div
            className="iblock pl15 verticalMiddle"
          >
            <p
              className="iblock f14 fw4 pfc3 pb2"
            >
              Filters
            </p>
            <p
              className="at_filterlength fw4 f10 pfc4"
            >
              0
               selected
            </p>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="iblock input-24 verticalMiddle"
            >
              <IconSvg />
            </span>,
            <div
              className="iblock pl15 verticalMiddle"
            >
              <p
                className="iblock f14 fw4 pfc3 pb2"
              >
                Filters
              </p>
              <p
                className="at_filterlength fw4 f10 pfc4"
              >
                0
                 selected
              </p>
            </div>,
          ],
          "className": "pl15 pb5 pt5 pr15 flex alignCenter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <IconSvg />,
              "className": "iblock input-24 verticalMiddle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="iblock f14 fw4 pfc3 pb2"
                >
                  Filters
                </p>,
                <p
                  className="at_filterlength fw4 f10 pfc4"
                >
                  0
                   selected
                </p>,
              ],
              "className": "iblock pl15 verticalMiddle",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Filters",
                  "className": "iblock f14 fw4 pfc3 pb2",
                },
                "ref": null,
                "rendered": "Filters",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    0,
                    " selected",
                  ],
                  "className": "at_filterlength fw4 f10 pfc4",
                },
                "ref": null,
                "rendered": Array [
                  0,
                  " selected",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
